lib/plug/conn.ex

```diff
@@ -1696,6 +1696,25 @@ defmodule Plug.Conn do
     conn |> get_session |> Map.get(session_key(key))
   end
 
+  @doc """
+  Returns session value for the given `key`.
+
+  Returns the default value if `key` does not exist.
+
+  The key can be a string or an atom, where atoms are
+  automatically converted to strings.
+  """
+  @spec get_session(t, String.t() | atom, any) :: any
+  def get_session(conn, key, default) when is_atom(key) or is_binary(key) do
+    session = get_session(conn)
+    key = session_key(key)
+
+    case Map.has_key?(session, key) do
+      false -> default
+      true -> Map.get(session, key)
+    end
```

Comment:
```suggestion
    conn |> get_session() |> Map.get(session_key(key), default)
```

In fact, you can probably merged this into the function before it by doing this:

```elkixir
def get_session(conn, key, default \\ nil) when is_atom(key) or is_binary(key) do
  conn |> get_session() |> Map.get(session_key(key), default)
end
```

:)

Reply:
Ha! Of course, we're dancing around the most obvious answer. 🔥 Good call, pushed!